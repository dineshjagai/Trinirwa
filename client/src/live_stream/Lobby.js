import React from 'react';
import NavBar from '../components/navBar';

const Lobby = ({
  username,
  roomName,
  handleRoomNameChange,
  handleSubmit,
  connecting,
}) => (
  <form onSubmit={handleSubmit}>
    <div>
      <NavBar />
      <div className="home">
        <div
          style={{
          }}
        />

        <div className="card shadow mx-auto mt-5" style={{ width: '30rem' }}>
          <div className="container mx-1">
            <div className="card-body p-xs">
              <h2 className="card-title text-center font-weight-bold">Live Stream</h2>
              <div className="form-group">
                <h2>Enter a room</h2>
                <div>
                  <label htmlFor="name">
                    {' '}
                    Hi,
                    {' '}
                    {username}
                    , please select a room to start your live stream!

                  </label>
                </div>
                <label htmlFor="room">Room name:</label>
                <input
                  type="text"
                  id="room"
                  placeholder="Enter Room Name"
                  value={roomName}
                  onChange={handleRoomNameChange}
                  readOnly={connecting}
                  required
                />
              </div>
              <div className="video-group">
                <button
                  type="submit"
                  className="btn btn-primary w-100"
                  disabled={connecting}
                >
                  {connecting ? 'Connecting' : 'Join'}
                </button>
              </div>

            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
);

export default Lobby;
