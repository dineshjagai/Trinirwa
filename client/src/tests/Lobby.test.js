import React from 'react';
import Lobby from '../live_stream/Lobby';
import { createMount } from '@material-ui/core/test-utils';


test('Test Lobby snapshot', () => {
  const handleS = () => {
      console.log('Room Change');
  };
  const handleR = () => {
      console.log('Room Change');
  };
  const component = createMount(<Lobby 
    username={'isimbib'}
    roomName={'First Name'}
    handleRoomNameChange={handleR}
    handleSubmit={handleS}
    connecting={true}
  />
  );
  expect(component).toMatchSnapshot();
});
