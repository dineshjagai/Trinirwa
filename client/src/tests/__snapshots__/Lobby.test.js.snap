// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Lobby snapshot 1`] = `[Function]`;
